Tests/Functional/Asynchronous/main.swift

```diff
@@ -0,0 +1,184 @@
+            ("test_whenTimeoutIsImmediate_butNotAllExpectationsAreFulfilled_fails", test_whenTimeoutIsImmediate_butNotAllExpectationsAreFulfilled_fails),
+        ]
+    }
+
+    func test_waitingForAnUnfulfilledExpectation_fails() {
+        expectationWithDescription("foo")
+        waitForExpectationsWithTimeout(0.2, handler: nil)
+    }
+
+    func test_waitingForUnfulfilledExpectations_outputsAllExpectations_andFails() {
+        expectationWithDescription("bar")
+        expectationWithDescription("baz")
+        waitForExpectationsWithTimeout(0.2, handler: nil)
+    }
+
+    func test_waitingForAnImmediatelyFulfilledExpectation_passes() {
+        let expectation = expectationWithDescription("flim")
+        expectation.fulfill()
+        waitForExpectationsWithTimeout(0.2, handler: nil)
+    }
+
+    func test_waitingForAnEventuallyFulfilledExpectation_passes() {
+        let expectation = expectationWithDescription("flam")
+        let timer = NSTimer.scheduledTimer(0.1, repeats: false) { _ in
+            expectation.fulfill()
+        }
+        NSRunLoop.currentRunLoop().addTimer(timer, forMode: NSDefaultRunLoopMode)
+        waitForExpectationsWithTimeout(1.0, handler: nil)
+    }
+
+    func test_waitingForAnExpectationFulfilledAfterTheTimeout_fails() {
+        let expectation = expectationWithDescription("hog")
+        let timer = NSTimer.scheduledTimer(1.0, repeats: false) { _ in
+            expectation.fulfill()
+        }
+        NSRunLoop.currentRunLoop().addTimer(timer, forMode: NSDefaultRunLoopMode)
+        waitForExpectationsWithTimeout(0.1, handler: nil)
+    }
+
+    func test_whenTimeoutIsImmediate_andAllExpectationsAreFulfilled_passes() {
+        let expectation = expectationWithDescription("smog")
+        expectation.fulfill()
+        waitForExpectationsWithTimeout(0.0, handler: nil)
+    }
+
+    func test_whenTimeoutIsImmediate_butNotAllExpectationsAreFulfilled_fails() {
+        expectationWithDescription("dog")
+        waitForExpectationsWithTimeout(-1.0, handler: nil)
+    }
+}
+
+class HandlerTestCase: XCTestCase {
+    static var allTests: [(String, HandlerTestCase -> () throws -> Void)] {
+        return [
+            ("test_whenExpectationsAreNotFulfilled_handlerCalled_andFails", test_whenExpectationsAreNotFulfilled_handlerCalled_andFails),
+            ("test_whenExpectationsAreFulfilled_handlerCalled_andPasses", test_whenExpectationsAreFulfilled_handlerCalled_andPasses),
+        ]
+    }
+
+    func test_whenExpectationsAreNotFulfilled_handlerCalled_andFails() {
+        self.expectationWithDescription("fog")
+
+        var handlerWasCalled = false
+        self.waitForExpectationsWithTimeout(0.2) { error in
+            XCTAssertNotNil(error, "Expectation handlers for unfulfilled expectations should not be nil.")
+            XCTAssertTrue(error!.domain.hasSuffix("XCTestErrorDomain"), "The last component of the error domain should match Objective-C XCTest.")
```

Comment:
Aside: it makes me sad that we have to force-unwrap here. It would be nice if `XCTAssertTrue` (and other assertions) would accept optional values and fail on nil (à la [Nimble](https://github.com/Quick/Nimble/blob/master/Tests/Nimble/Matchers/BeLogicalTest.swift#L36))